### ISSUE TITLE
`line_segment_transfer_error_one_way` is distance × segment length, so RANSAC's line inlier threshold depends on length

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4867]
`line_segment_transfer_error_one_way(ls1, ls2, H)` maps both image-1 endpoints with `H` and returns the mean of `|l2 · x|`, where `l2 = s2 × e2` is the cross product of the homogeneous image-2 endpoints and is never normalised. For that line, `|l2 · x|` equals the perpendicular distance times the image-2 segment length `L = ‖e2 − s2‖`, so the result is not a distance: a 3-px offset scores 197 on a 66-px segment and 994 on a 331-px segment. `RANSAC("homography_from_linesegments")` compares this value with `inl_th ** 2` (the docstring says `inl_th` is the "threshold for the correspondence to be an inlier. Internally is squared"). A segment offset by d px is therefore an inlier only when `d · L ≤ inl_th²`. With a 3-px offset, a point match becomes an inlier at `inl_th = 3`, but a segment needs `inl_th ≈ 14` at L = 66 px and more than 30 at L = 331 px. At the default `inl_th = 2.0`, a 0.05-px offset is rejected on every segment longer than 80 px. The docstring calls the result a "point-to-line error" and "the computed distance"; a point-to-line distance in pixels is `|l · x| / √(a² + b²)` for `l = (a, b, c)`.

## Repro

```python
import torch
from kornia.geometry.homography import line_segment_transfer_error_one_way
from kornia.geometry.linalg import transform_points
from kornia.geometry.ransac import RANSAC

H = torch.tensor([[[1.034685, -0.04160598, -63.61482],
                   [0.1039753, 1.135451, -62.87412],
                   [2.851519e-04, 1.404178e-04, 1.0]]], dtype=torch.float64)
# four image-1 segments of different lengths; image-2 segments = H(image-1 endpoints), exact
ls1 = torch.tensor([[[160.0, 95.0], [230.0, 96.0]],
                    [[160.0, 95.0], [330.0, 150.0]],
                    [[160.0, 95.0], [440.0, 320.0]],
                    [[230.0, 320.0], [440.0, 150.0]]], dtype=torch.float64)
ls2 = transform_points(H, ls1.reshape(1, 8, 2))[0].reshape(4, 2, 2)
p1, p2 = ls1[:, 0], ls2[:, 0]          # the four start points, as point matches
rp, rl = RANSAC("homography"), RANSAC("homography_from_linesegments")

d = 3.0
print(f"every correspondence displaced by {d} px in image 2 (points: along x; segments: perpendicular to themselves)")
v = ls2[:, 1] - ls2[:, 0]
L = v.norm(dim=-1)
n = torch.stack([-v[:, 1], v[:, 0]], -1) / L[:, None]
ls2_bad = ls2 + d * n[:, None, :]
p2_bad = p2 + torch.tensor([d, 0.0], dtype=torch.float64)
err = line_segment_transfer_error_one_way(ls1[None], ls2_bad[None], H)[0]
for k in range(4):
    print(f"  segment {k}: image-2 length L = {L[k]:6.1f} px -> line error {err[k]:7.2f} (d*L = {d * L[k]:7.2f})")
# RANSAC.forward calls verify(..., self.inl_th ** 2); run it with the true H to isolate the threshold
for th in (2.0, 2.99, 3.01, 10.0, 15.0, 20.0, 25.0, 30.0):
    ip = rp.verify(p1, p2_bad, H, th**2)[1].tolist()
    il = rl.verify(ls1, ls2_bad, H, th**2)[1].tolist()
    print(f"  inl_th = {th:5}: points inlier {ip} | segments inlier {il}")
print("default inl_th = 2.0, perpendicular shift 0.05 px:",
      rl.verify(ls1, ls2 + 0.05 * n[:, None, :], H, 2.0**2)[1].tolist())
```

```text
every correspondence displaced by 3.0 px in image 2 (points: along x; segments: perpendicular to themselves)
  segment 0: image-2 length L =   65.7 px -> line error  197.11 (d*L =  197.11)
  segment 1: image-2 length L =  166.0 px -> line error  498.00 (d*L =  498.00)
  segment 2: image-2 length L =  331.2 px -> line error  993.60 (d*L =  993.60)
  segment 3: image-2 length L =  248.3 px -> line error  744.87 (d*L =  744.87)
  inl_th =   2.0: points inlier [False, False, False, False] | segments inlier [False, False, False, False]
  inl_th =  2.99: points inlier [False, False, False, False] | segments inlier [False, False, False, False]
  inl_th =  3.01: points inlier [True, True, True, True] | segments inlier [False, False, False, False]
  inl_th =  10.0: points inlier [True, True, True, True] | segments inlier [False, False, False, False]
  inl_th =  15.0: points inlier [True, True, True, True] | segments inlier [True, False, False, False]
  inl_th =  20.0: points inlier [True, True, True, True] | segments inlier [True, False, False, False]
  inl_th =  25.0: points inlier [True, True, True, True] | segments inlier [True, True, False, False]
  inl_th =  30.0: points inlier [True, True, True, True] | segments inlier [True, True, False, True]
default inl_th = 2.0, perpendicular shift 0.05 px: [True, False, False, False]
```

kornia `main` 7ddf731bd, torch 2.14.0 (and 2.5.1: same output), CPU float64, macOS arm64.

`RANSAC.forward` scores models with `self.verify(kp1, kp2, models, self.inl_th**2)`; the repro calls `verify` with the true `H` so that local optimisation cannot move the model. `find_homography_lines_dlt_iterated` feeds the same length-scaled value into its `exp(-e / (2 soft_inl_th²))` weights, so its soft threshold depends on segment length too. This was ruled a wart on 2026-09-24 in the conventions batch 8 audit.

## Classification

Silent wrong result: the inlier set of a line-segment RANSAC depends on segment length, and no single `inl_th` treats a short segment and a long segment alike.

Idiom 3 (a guard whose predicate does not match the documented contract): the inlier test `error ≤ inl_th²` is documented as a distance threshold, but the error it tests is in pixels times the segment length in pixels.

## Disposition

MERGE IN WINDOW — behaviour change: the line error has to be a pixel distance before any threshold on it means anything. It becomes the mean endpoint-to-line distance, `|l2 · x|` divided by `‖(a, b)‖` of `l2`, and RANSAC compares it with `inl_th` on the same footing as the point error (the square of the distance against `inl_th²`). Squaring the current value alone would not fix it, because the value would still grow with L². The values `line_segment_transfer_error_one_way` returns change by a factor of 1/L. Any line-segment `RANSAC` or `find_homography_lines_dlt_iterated` threshold tuned to the current values selects different inliers and has to be re-tuned in pixels.

Found by the batch-8 conventions audit (two-view geometry); a `test_wart_*` pin in the docs PR flips when the fix lands.

Related: #4866 (the line DLT these models come from); #4870 (the same exp kernel in the point IRLS).

Posted on behalf of @ducha-aiki by Claude (Fable 5.1).
